src/lib.rs

```diff
@@ -16,38 +16,53 @@
 //! In order to encrypt and decrypt data, we needs to generate a keypair, i.e. a secret key and a public key.
 //! ```
 //! let scheme = cupcake::default();
-//! use cupcake::traits::{SKEncryption, PKEncryption};
+//! use cupcake::traits::{KeyGeneration};
 //! let (pk, sk) = scheme.generate_keypair();
 //! ```
 //! The public key can be used for encryption and the secret key can be used for encryption or decryption.
 //!
 //! # Encryption and Decryption
 //!
-//! The library currently supports one plaintext type, which is `vec<u8>` of fixed size n. We can encrypt a vector under a public key like so
+//! The default plaintext space of Cupcake is `Vec<u8>` of fixed size n. We can encrypt a vector under a public key like so
 //! ```
 //! # let scheme = cupcake::default();
-//! # use cupcake::traits::{SKEncryption, PKEncryption};
+//! # use cupcake::traits::KeyGeneration;
 //! # let (pk, sk) = scheme.generate_keypair();
+//! use cupcake::traits::{SKEncryption, PKEncryption};
 //! let v = vec![1; scheme.n];
 //! let ct = scheme.encrypt(&v, &pk);
 //! ```
 //! Then, the ciphertext `ct` can be decrypted using the secret key:
-//!
 //! ```
 //! # let scheme = cupcake::default();
-//! # use cupcake::traits::{SKEncryption, PKEncryption};
+//! # use cupcake::traits::{KeyGeneration, SKEncryption, PKEncryption};
 //! # let (pk, sk) = scheme.generate_keypair();
 //! # let v = vec![1; scheme.n];
 //! # let ct = scheme.encrypt(&v, &pk);
-//! let w = scheme.decrypt(&ct, &sk);
+//! let w: Vec<u8>= scheme.decrypt(&ct, &sk);
 //! assert_eq!(v, w);
 //! ```
+//! You may also use other plaintext types, which are vectors of `Scalar` of size n. Here `Scalar`  represents an integer modulo a fixed modulus t.
+//! See the following example:
+//! ```
+//! # use cupcake::traits::{KeyGeneration, PKEncryption, SKEncryption};
+//! use cupcake::integer_arith::scalar::Scalar;
+//! use cupcake::integer_arith::ArithUtils;
```

Comment:
See if this can be simplified. Maybe move conversion to  a standard trait and do the mod t automatically in encryption